src/WebJobs.Extensions.DurableTask/LinuxAppServiceLogger.cs

```diff
@@ -57,12 +53,26 @@ internal class LinuxAppServiceLogger
         {
             // Initializing fixed logging metadata
             this.writeToConsole = writeToConsole;
-            this.roleInstance = JToken.FromObject("App-" + containerName);
-            this.tenant = JToken.FromObject(tenant);
-            this.stamp = stampName;
 
-            var finalCharIndex = stampName.Length - 1;
-            this.primaryStamp = char.IsLetter(stampName[finalCharIndex]) ? stampName.Remove(finalCharIndex) : stampName;
+            // Since the values below are obtained via a NameResolver, they might be null.
+            // Attempting to serialize a null value results in exceptions, or even worse, wrong logs,
+            // so we need to be careful.
+            if (!string.IsNullOrEmpty(containerName))
+            {
+                this.roleInstance = JToken.FromObject("App-" + containerName);
```

Comment:
Is there a reason we are using JToken.FromObject here instead of just explicitly adding the string?

Reply:
It's just an optimization, since internally JObjects only store JTokens, I'm just doing that work in advance. Might be a little overkill, happy to revert to strings only

Reply:
I would recommend against this. Vanilla strings are much easier to debug and parsing to JToken early like this is likely to confuse the future readers of this code because it doesn't map to any obvious intent. Generally speaking, if you're doing a performance optimization that might not be obvious to the reader, then you should include a comment explaining why you're doing what you're doing.

Reply:
Yeah you're right. I'll remove it, was a silly thing to do

Reply:
done!